DOCSUP-13875: Document the replication_wait_for_inactive_replica_timeout setting (#28464)
ClickHouse/ClickHouse: docs/en/operations/settings/settings.md

-   1 — Projection optimization is obligatory.

Default value: `0`.

## replication_alter_partitions_sync {#replication-alter-partitions-sync}

> Allows to specify waiting for actions to manipulate the partitions.

Reviewer: ```suggestion
Allows to specify the delay for actions that manipulate the partitions.
```

Possible values:

> -   0 — Do not wait.
> -   1 — To wait for own execution.
> -   2 — To wait for everyone.

Reviewer: ```suggestion
-   0 — Do not wait. Means that the next action can start any time, not waiting for the previous action to be completed.
-   1 — Wait for own execution. Means that the next action on the partition starts only after the previous action is completed on this partition.
-   2 — Wait for everyone. Means that the next action starts only after the previous action is completed on all partitions.
```

Default value: `1`.

## replication_wait_for_inactive_replica_timeout {#replication-wait-for-inactive-replica-timeout}

> Allows to specify how long (in seconds) to wait for inactive replicas to execute [ALTER](../../sql-reference/statements/alter/index.md), [OPTIMIZE](../../sql-reference/statements/optimize.md) or [TRUNCATE](../../sql-reference/statements/truncate.md) queries.

Reviewer: ```suggestion
Specifies how long (in seconds) to wait for inactive replicas to execute [ALTER](../../sql-reference/statements/alter/index.md), [OPTIMIZE](../../sql-reference/statements/optimize.md) or [TRUNCATE](../../sql-reference/statements/truncate.md) queries.
```

Possible values:

-   0 — Do not wait.

> -   Negative integer — To wait for unlimited time.
> -   Positive integer — The number of seconds of waiting.

Reviewer: ```suggestion
-   Negative integer — Wait for unlimited time.
-   Positive integer — The number of seconds to wait.
```

Default value: `0`.

## replication_alter_partitions_sync {#replication-alter-partitions-sync}

Determines how actions that manipulate partitions are executed.

Possible values:

-   0 — Do not wait. The next action can start any time, not waiting for the previous action to be completed.
-   1 — Wait for own execution. The next action on the partition starts only after the previous action is completed on this partition.

> -   2 — Wait for everyone. The next action starts only after the previous action is completed on all partitions.

Reviewer 2: What is "the next action" and "the previous action"?

> -   0 — Do not wait. Means that the next action can start any time, not waiting for the previous action to be completed.
> -   1 — Wait for own execution. Means that the next action on the partition starts only after the previous action is completed on this partition.
> -   2 — Wait for everyone. Means that the next action starts only after the previous action is completed on all partitions.

Reviewer: ```suggestion
-   0 — Do not wait. The next action can start any time, not waiting for the previous action to be completed.
-   1 — Wait for own execution. The next action on the partition starts only after the previous action is completed on this partition.
-   2 — Wait for everyone. The next action starts only after the previous action is completed on all partitions.
```

Default value: `1`.

## replication_wait_for_inactive_replica_timeout {#replication-wait-for-inactive-replica-timeout}

Specifies how long (in seconds) to wait for inactive replicas to execute [ALTER](../../sql-reference/statements/alter/index.md), [OPTIMIZE](../../sql-reference/statements/optimize.md) or [TRUNCATE](../../sql-reference/statements/truncate.md) queries.

Possible values:

-   0 — Do not wait.
-   Negative integer — Wait for unlimited time.
-   Positive integer — The number of seconds to wait.

> Default value: `120`.

Reviewer: ```suggestion
Default value: `120` seconds.
```
